### ISSUE TITLE
C function cropped in member group

### ISSUE THREAD

**Issue by crasbe** [doxygen/doxygen#7413]
Hello,

I have noticed an issue regarding #defines in member groups. When a block of one or more `#defines` is inside a member group and one or more functions is inside another member group, the source code in the resulting documentation is cropped.

Doxyfile options:
- Optimize for C or PHP Output
- HTML output -> prepare for compressed HTML (.chm)
- EXTRACT_ALL activated
- EXTRACT_STATIC activated
- INLINE_SOURCES activated
- SEARCHENGINE deactivated

Doxygen Version is 1.8.16, Graphviz is Version 2.38 and the Microsoft HTML Help Workshop is Version 1.32.

Attached are the testfiles and the Doxyfile in a zip archive as well as screenshots of the resulting HTML pages.

Best Regards

Attachments:

This is the expected behaviour, when the member group around the #define is deleted (Testfile_good.h):
![expected_behaviour](https://user-images.githubusercontent.com/1836007/69244809-f2d86a00-0ba5-11ea-95b1-b1409d34f76d.PNG)

This is the behaviour from Testfile_bad.h:
![unexpected_behaviour](https://user-images.githubusercontent.com/1836007/69244813-f3710080-0ba5-11ea-936b-54699ef90ffd.PNG)

Source Code of Testfile_good.h:
```C
/** @name Macrotest
 *  @{
 */
#define asdf 1
/** @}
 */

/** @name Function Test
 *  @{
 */
inline static void testfunc(void) 
{
	void* voidptr;
}
/** @}
 */
```

Source Code of Testfile_bad.h:
```C
#define asdf 1

/** @name Function Test
 *  @{
 */
inline static void testfunc(void) 
{
	void* voidptr;
}
/** @}
 */
```
[Doxygen Bug.zip](https://github.com/doxygen/doxygen/files/3869635/Doxygen.Bug.zip)

**Comment by albert-github** [comment]
With the version 1.8.15 I do see the problem as indicated, i.e. :
![bad_1_8_16](https://user-images.githubusercontent.com/5223533/69247538-cffc8480-0baa-11ea-96b7-2df8205f601b.jpg)

with the current master version(2972f6ddc491f615f814b1277ec715b6f707c419) I see:
![bad_master](https://user-images.githubusercontent.com/5223533/69247587-e571ae80-0baa-11ea-8bbe-af7f2a0c8aa8.jpg)

So I think the problem has been fixed but not yet release,

**Comment by doxygen** [comment]
This issue was previously marked 'fixed but not released',
which means it should be fixed in doxygen version 1.8.17.
Please verify if this is indeed the case. Reopen the
issue if you think it is not fixed and please include any additional information
that you think can be relevant (preferably in the form of a self-contained example).
